src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/sentencing/SentencingAdjustmentsService.kt

```diff
@@ -132,6 +132,34 @@ class SentencingAdjustmentsService(
       ?: throw RuntimeException("No mapping found for adjustment ${createEvent.additionalInformation.id}, maybe we never received a create")
   }
 
+  suspend fun deleteAdjustment(createEvent: AdjustmentDeletedEvent) {
+    sentencingAdjustmentsMappingService.getMappingGivenAdjustmentId(createEvent.additionalInformation.id)
+      ?.also { mapping ->
+        if (mapping.nomisAdjustmentCategory == "SENTENCE") {
+          nomisApiService.deleteSentenceAdjustment(
+            mapping.nomisAdjustmentId,
+          )
+        } else {
+          nomisApiService.deleteKeyDateAdjustment(
+            mapping.nomisAdjustmentId,
+          )
+        }.also {
+          sentencingAdjustmentsMappingService.deleteMappingGivenAdjustmentId(createEvent.additionalInformation.id)
```

Comment:
if the mapping deletion fails - will it go back on the DLQ and attempt the nomis deletion again?

Reply:
yes - a test sounds good